common/models/user.js

```diff
@@ -673,6 +673,49 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    UserModel.observe('before save', function beforeEmailUpdate(ctx, next) {
+      if (ctx.isNewInstance) return next();
+      if (ctx.where) {
+        ctx.Model.find({ where: ctx.where }, function(err, userInstances) {
+          if (err) return next(err);
+          getOriginalUserData(userInstances);
+        });
+      } else if (ctx.instance) {
+        ctx.hookState.id = ctx.instance.id;
+        var oldId = ctx.hookState.id;
+        ctx.Model.find({ where: { id: oldId }}, function(err, userInstances) {
+          if (err) return next (err);
+          getOriginalUserData(userInstances);
+        });
+      }
+
+      function getOriginalUserData(userInstances) {
+        ctx.hookState.originalUserData = userInstances.map(function(u) {
+          return { id: u.id, email: u.email };
+        });
+      }
+      next();
```

Comment:
Once again. This `next` is misplaced, you need to call it from inside `Model.find` callback.

Considering that both branches use the same callback, I am suggesting the following implementation:

```js
var where = ctx.where || { id: ctx.instance.id }; // feel free to add more safety checks
ctx.Model.find({ where: where }, function(err, userInstances) {
  if (err) return next(err);
  ctx.hookState.originalUserData = userInstances.map(function(u) {
    return { id: u.id, email: u.email };
  });
  next();
});
```